enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -1743,6 +1797,7 @@ func (s *SchedulerServer) enqueueTaskReservations(ctx context.Context, enqueueRe
 			if len(candidateNodes) == 0 {
 				return errTaskSizeTooLarge(pool, os, arch, enqueueRequest.GetTaskSize())
 			}
+			candidateNodes = weightedResample(candidateNodes)
```

Comment:
`task_router.RankNodes()` does a random shuffle on the non-preferred nodes, so it will undo the work done here - probably the task_router needs to be in charge of this?

Reply:
I don't think the order matters in this case, it's just the population of nodes that I'm after. LMK if you still think this is wrong.